lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
+        :type     => :certificate_revocation,
+        :desc     => <<EOT
+Whether certificate revocation checking should be enabled, and what level of checking should be performed.
+
+When certificate_revocation is set to 'true' or 'chain', Puppet will download the CA CRL and will perform revocation
+checking against each certificate in the chain. When Puppet is using a single root certificate and has no intermediates
```

Comment:
Whether we should update this comment depends on if/when we're going to fix chain CRL verification, and whether we need to expose the details of CRL checking to most users who will be using a root CA certificate. My approach was that this behavior has been an issue since CRL checking was implemented so we should generally preserve the existing code paths, while optimistically planning for actually fixing this. if it's likely that we're not going to implement chain verification then we should add the caveats that you've written.